tfhe/src/zk.rs

```diff
@@ -123,12 +125,30 @@ impl ZkVerificationOutCome {
     }
 }
 
+#[derive(Clone, Copy, Debug, PartialEq, Eq)]
+pub struct ZkMSBZeroPaddingBitCount(pub u64);
+
+/// The CRS (Common Reference String) of a ZK scheme is a set of values shared between the prover
+/// and the verifier.
+///
+/// The same CRS should be used at the prove and verify steps.
+#[derive(Clone, Debug, Serialize, Deserialize, Versionize)]
+#[repr(transparent)]
 pub struct CompactPkeCrs {
     public_params: CompactPkePublicParams,
 }
 
-#[derive(Clone, Copy, Debug, PartialEq, Eq)]
-pub struct ZkMSBZeroPaddingBitCount(pub u64);
+impl Named for CompactPkeCrs {
+    const NAME: &'static str = "zk::CompactPkeCrs";
+}
```

Comment:
isn't this an issue for versioning if the struct is transparent ?

Reply:
No, the name is included outside of the versioned data so it has no influence on the versioning. If one day we want to modify the struct to not be transparent we will have to do a CompactPkeCrsVersions enum that starts at v1:
```rust
enum CompactPkeCrsVersions {
  V0(CompactPkePublicParams),
  V1(CompactPkeCrs)
}
```

Reply:
ok my bad I meant if we want to safe deserialize a CompactPkeCrs from the params, it won't work because of the named impl right ?

Reply:
yes, the idea is to deserialize it as a CompactPkePublicParams and then convert it to a CompactPkeCrs using the From impl

Reply:
could we somehow force people to always deserialize the CRS via some from/into tricks ?

Reply:
The problem is that since they have been serialized with NAME = "CompactPkePublicParams" it means that the output of safe_deserialize needs to be something with NAME= "CompactPkePublicParams". 

We could have CompactPkeCrs::NAME = "CompactPkePublicParams", but since it's just a matter of calling `.into` I don't think it's worth having this discrepancy.

Reply:
no yeah I agree, was just thinking out loud

Reply:
or maybe a different way to write safe serialize/deserialize with a serialized type S that has Into<T> where T is the actual type we want to use later, but that would maybe defeat the purpose of the safe serialize